probe-rs/src/debug/registers.rs

```diff
@@ -117,14 +161,32 @@ impl DebugRegisters {
                         .find(|debug_register| debug_register.id == platform_register.id)
                     {
                         // Some register definitions are descriptive for registers defined with the same [`RegisterId`] elsewhere, so we treat them differently.
-                        special_register.special_name = Some(platform_register.name);
+                        special_register.alias = Some(platform_register.name);
                     } else {
                         // It is safe for us to push a new [`DebugRegister`]
                         debug_registers.push(DebugRegister {
                             register_file,
                             group: register_group,
                             name: platform_register.name(),
-                            special_name: None,
+                            alias: None,
+                            preserve_rule: match core.core_type() {
+                                CoreType::Armv6m | CoreType::Armv7em | CoreType::Armv7m => {
```

Comment:
Why is Armv8m excluded here?

Reply:
https://github.com/probe-rs/probe-rs/pull/1495#discussion_r1099174952